src/interfacers/EmonHubEmoncmsHTTPInterfacer.py

```diff
@@ -57,7 +59,17 @@ def _process_post(self, databuffer):
 
         # Construct post_url (without apikey)
         post_url = self._settings['url']+'/input/bulk'+'.json?apikey='
-        post_body = "data="+data_string+"&sentat="+str(sentat)
+        
+        # Construct post body
+        post_body_data = {"data": data_string, "sentat": sentat}
+        
+        if self._settings['compress']:
+            # Compress data and encode as hex string.
+            post_body_data["data"] = zlib.compress(post_body_data["data"]).encode("hex")
```

Comment:
This is throwing away the `sentat` key. I don't know if that's important.

More importantly this needs a bit of tweaking to make it work in Python3. The argument to `zlib.compress` must be bytes, so we need to `encode` it. `compress` then returns more bytes, but you need to use the `codecs` module to encode as hex.

However, hex encoding is doubling the size of the input, so that's not really what you want if the goal is to save space. Isn't urlencoding (which you're doing below) already sufficient?

Anyway, we're using requests now, which can handle binary POST, so there's no need for hex encoding or urlencoding. Hopefully php can also handle that. I guess it also assumes UTF-8 encoding?